Adding recorded tests for Server Trust Groups .NET client

bojovicmatija opened the pull request:
Adding tests for .NET cliennt for Server Trust Groups.

Commit:
addding recorded tests for Server Trust Groups

pakrym committed:
Prepare to produce a nightly build of EventHubs extension (#17520)

We need a package ref to Microsoft.Extensions.Azure

azabbasi committed:
ADT - livetest - Wait 5 minutes after role assignment and before the test run to avoid 403s. (#17539)

hivyas committed:
[Media] Removing periods from safename (#17537)

kasobol-msft committed:
[Storage] Bump threadpool in tests. (#17547)

dochung4 committed:
Make DedicatedHostGroup.properties.supportAutomaticPlacement default to false (#17224)

pakrym committed:
Fix Microsoft.Azure.WebJobs.Extensions.EventHubs inheritdoc failures (#17551)

Mohit-Chakraborty committed:
Generate 'SourcePath' attribute in Perf test assemblies (#17545)

* Generate 'SourcePath' attribute in Perf test assemblies

* Include stress projects in the condition to add the attribute

JoshLove-msft committed:
Include Lease Info in logging (#17509)

Fixes https://github.com/Azure/azure-sdk-for-net/issues/17032

JoshLove-msft committed:
Ensure that type is populated when deserializing into a CloudEvent (#17468)

During UX studies, users were easily confused if they attempted to deserialize an event grid event into a CloudEvent as no exception was thrown. We intentionally removed the validation as part of addressing https://github.com/Azure/azure-sdk-for-net/issues/16012, but this may have gone too far in the other direction. A good compromise seems to leave in the validation that the service uses which is to ensure that Type is populated for CloudEvents, while allowing Source to be null.

pakrym committed:
Remove explicit package references on net5.0 (#17544)

crtreasu-msft committed:
Added pipelines for Mixed Reality SDKs (#17554)

* This change adds pipelines for Mixed Reality services

* Fixed PR path

* Removed TestPipeline: true

Co-authored-by: Craig Treasure <crtreasu@microsoft.com>

pakrym committed:
Fix the flakiness in secrets extension test (#17559)

sasapopo commented:
#assign sasapopo

aim-for-better committed:
[HDInsight] Support extending list hosts api and computeisolation properties. (#17532)

* Support extending list hosts api and computeisolation properties.

* Fix test errors

Co-authored-by: Zhenyu Zhou <zhezhou@microsoft.com>

pavelprystinka committed:
[Communication] - PhoneNumberAdministration - add tests (#17492)

* Add PhoneNumberAdministration tests

pakrym committed:
Prepare Azure.Extensions.AspNetCore.DataProtection.Blobs for release (#17570)

ramya-rao-a committed:
[Service Bus] Use docs.microsoft.com link for API references (#17577)

azure-sdk committed:
Fixed the bug of replacing img src with href text (#17586)

Co-authored-by: Sima Zhu <sizhu@microsoft.com>

minnieliu committed:
[Communication] - All - Rename CommunicationUserCredential to CommunicationTokenCredential (#17553)

* Rename CommunicationUserCredential to CommunicationTokenCredential

* Update changelog

* Update readme

* Update readme

* Renaming bearer test file

Co-authored-by: Minnie Liu <peiliu@microsoft.com>

chamons committed:
[synapse] Add two new champion scenario tests, refactor one, and disable a few (#17546)

- Add beginnings of two new champion scenario tests (Upload artifact notebook and monitoring pipelines)
  - Once all of the tests are in they will be turned into snippets and made markdown
  - Significant amount of polling code in Sample2_ExecuteSparkStatement.cs is from service team recommendation right now
      - Moving it into the binding as a LRO is a future research topic
      - Tracked in https://github.com/Azure/azure-sdk-for-net/issues/17587 to track the copied code. 
- Refactor CreateAndRunPipeline test which had two tests, which were order dependent :facepalm: 
- Disable a few more tests that now fail in LIVE (will be reenabled when the [swagger PR](https://github.com/Azure/azure-rest-api-specs/pull/12093) lands and is bumped to)

azure-sdk committed:
Increment version for extensions releases (#17592)

Increment package version after release of Azure.Extensions.AspNetCore.DataProtection.Blobs

seanmcc-msft committed:
Feature/storage/data lake set permission bug (#17591)

jaschrep-msft committed:
Fixed broken docs link, (#17593)

Points developers to the overload we want them to use, rather than the
hidden one we "retired."
Likely solves an MS Docs rendering bug, where it can't link to the
method properly because methods we hide from intellisense also aren't
rendered to MS Docs at all, so there's nothing to link to.

Co-authored-by: jschrepp-MSFT <41338290+jschrepp-MSFT@users.noreply.github.com>

dhawal777 committed:
Updating release notes. (#17525)

* Updating release notes to highlight additional changes that came as part of last update

* Spell fix

seanmcc-msft committed:
Fixed bug where BlobBaseClient.Exists() did not function correctly for blob encrypted with Customer Provided Key or Encryption Scope. (#17599)

jaschrep-msft committed:
Misc blobs migration guide fixes (#17601)

* Misc blobs migration guide fixes

Fixed some reported ambiguities and rendering bugs.

* Fixed variable name in sample project

* build error

Co-authored-by: jschrepp-MSFT <41338290+jschrepp-MSFT@users.noreply.github.com>

JoshuaLai committed:
[Communication] - Identity - Auth with Token Credential (#17438)

* [CommunicationIdentityClient] adding in new method to create beaer token. Fixing spelling mistake.

* [CommunicationIdentityClient] adding new init passing in token credential

* [TokenCredential] finalize the parameters to take in a token credential and an endpoint string

* [ClientOptionsExtensions] removing the scope from the BearerTokenAuth policy

* [CommunicationIdentityClientLiveTests] Adding new live test init with token credential

* [PhoneNumberAdministrationClient] revert the init with token

* [CommunicationIdentityClient] passing in endpoint as an uri

* [CommunicationIdentityClient] assert on the endpoint

* [CommunicationIdentityClient] reorder of the parameters

* Adding working live tests

* [Scripts] New generated code

* [CommunicationIdentityClient] update the comment for tokenCredential to be accurate

* [CommunicationIdentityClient] update the comment for endpoont to be more accurate

* [CommunicationIdentityClientLiveTests] Nit on the the test name

* Update CHANGELOG.md

* [IdentityClientFromTokenWithMultipleScopes] rename the generated files

* [CommunicationRecordedTestSanitizer] adding sanitizer to COMMUNICATION_ENDPOINT_STRING

Co-authored-by: Minnie Liu <peiliu@microsoft.com>

pakrym committed:
Auto create storage container (#17607)

pakrym committed:
Organize Packages.Data.props (#17603)

This PR splits `Packages.Data.props` into strict groups:

1. Track 1 dependencies
2. Track 2 dependencies
3. Track 2 extended dependencies (Microsoft.Azure.* T2 libraries)
4. Build time dependencies
5. Test dependencies

Some versions are duplicated across groups, this is intentional so someone trying to update test package version doesn't change the shipping package dependency.

pakrym committed:
Prepare Azure.Extensions.AspNetCore.DataProtection.Blobs for release again (#17610)

pakrym committed:
Add EventHub extension README.md (#17514)

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/16646

Commit:
rerun test

Commit:
rerun test

Commit:
adding correct session records

Commit:
adding correct session records

AlexGhiondea committed:
Update CODEOWNERS (#17605)

The TimeSeriesInsights label was incorrect. Fixing it.

Also ensure that we label PRs in the Synapse folder as well.

SanjaMalesevic reviewed (commented):
1. Clients for server trust groups have already been generated?
2. I don't think there is a need to single out tests where you retrieve groups by location and by instance? I would add that to the first test as well

SanjaMalesevic commented on sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Microsoft.Azure.Management.Sql.csproj:
```diff
@@ -7,7 +7,7 @@
 		<Description>Azure SQL Management SDK library</Description>
 		<AssemblyName>Microsoft.Azure.Management.Sql</AssemblyName>
-		<Version>1.47.0-preview</Version>
+		<Version>1.47.1-preview</Version>
```
maybe 1.48.0-preview here?

SanjaMalesevic commented on sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Properties/AssemblyInfo.cs:
```diff
@@ -22,5 +22,5 @@
 [assembly: AssemblyDescription("Provides management functionality for Microsoft Azure SQL.")]
 [assembly: AssemblyVersion("1.0.0.0")]
-[assembly: AssemblyFileVersion("1.47.0.0")]
+[assembly: AssemblyFileVersion("1.47.1.0")]
```
same as above

azure-sdk committed:
Increment version for extensions releases (#17627)

Increment package version after release of Azure.Extensions.AspNetCore.DataProtection.Blobs

JoshLove-msft committed:
Authorization refresh fix (#17609)

* save

* Authorization refresh fix

* Undo inadvertent changes

minnieliu committed:
[Communication] - Common - Adding Identifier Suffix to all Identifier types (#17608)

* Adding Identifier Suffix to all Identifier types

* Update generated files

* Fix minor issues

* Update changelog

Co-authored-by: Minnie Liu <peiliu@microsoft.com>

JoshuaLai committed:
[COMMUNICATION] - [LIVE TESTS] - adding missing env var (#17631)

* [test-resource] adding in new environment variable that was missed for CI live test

* [resource] encoded space

Mohit-Chakraborty committed:
Log exceptions from the failed step in perf test environment (#17629)

OlegSternbergIBM committed:
Audit devops sdk (#17529)

* generated sdk + tests

* Update version

* Stage test passes. Both files need to be adjusted for Prod, once API will be exposed there.

* consiume new DevOps swagger

* validated DevOps in Prod

* fix version

* update recorded test file

* updated version to 1.48

Co-authored-by: Oleg Sternberg <olsternb@microsoft.com>

crtreasu-msft committed:
Added the Azure.MixedReality.Authentication library (#17571)

This change adds an authentication library for Mixed Reality services, which all currently utilize a custom STS for authentication. Upcoming Mixed Reality client libraries will depend on this library to perform authentication using identities from Azure.Identity. See the added `README.md` for more information about the library.

[ApiView link](https://apiview.dev/Assemblies/Review/35afdb6afa874f6ab4caa1cfaf7a15cd)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 1287 (#17630)

* Move common code to create API review into eng common in tools

azure-sdk committed:
Update AutoRest C# version (#17516)

Update AutoRest C# version to 3.0.0-beta.20201217.1

pakrym committed:
Add System.IdentityModel.Tokens.Jwt to versions (#17637)

chjaress committed:
Add tests for enableAutomaticUpgrade property for VM Extensions (#17604)

* Adding VM Extension test and session records for tests

* Adding test for VMScaleSetVMExtensions

* Reverting some generated files

ShivangiReja committed:
Update "AutoRest.CSharp.V3" package name to "AutoRest.CSharp" (#17638)

Update "AutoRest.CSharp.V3" package name to "AutoRest.CSharp"

pakrym committed:
Add AsyncPageable scope validation (#17635)

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/17633

Commit:
addding recorded tests for Server Trust Groups

# Conflicts:
#	sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Microsoft.Azure.Management.Sql.csproj
#	sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Properties/AssemblyInfo.cs

Commit:
Merge branch 'azureNetSTGTests' of https://github.com/bojovicmatija/azure-sdk-for-net into azureNetSTGTests

# Conflicts:
#	sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Microsoft.Azure.Management.Sql.csproj
#	sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Properties/AssemblyInfo.cs

bojovicmatija commented:
> 1. Clients for server trust groups have already been generated?
> 2. I don't think there is a need to single out tests where you retrieve groups by location and by instance? I would add that to the first test as well

sopanic commented on sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/ServerTrustGroupTests.cs:
```diff
@@ -0,0 +1,153 @@
+namespace Sql.Tests
+{
+	/// <summary>
+	/// Contains tests for the lifecycle of a sanaged instance and managed database security alert policy
```
Update the summary. Also, you can maybe add a summary to test cases as well.

sopanic commented on sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/ServerTrustGroupTests.cs:
```diff
@@ -0,0 +1,153 @@
+
+			Sku sku = new Sku();
+			sku.Name = "MIGP8G4";
+			sku.Tier = "GeneralPurpose";
```
Are there any constants that can be used here, instead of strings?

sopanic commented on sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/ServerTrustGroupTests.cs:
```diff
@@ -0,0 +1,153 @@
+
+				ResourceGroup resourceGroup = context.CreateResourceGroup(TestEnvironmentUtilities.DefaultLocationId);
+				Assert.NotNull(resourceGroup);
+				createManagedInstances(context, resourceGroup, managedInstanceNames);
```
Consider putting this in a separate Setup method, as it's used in all tests cases in the same manner. Same goes for deleting managed instances - Cleanup method.

sopanic commented on sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/ServerTrustGroupTests.cs:
```diff
@@ -0,0 +1,153 @@
+				ServerTrustGroup parameters = new ServerTrustGroup(groupMembers, trustScopes);
+
+				ServerTrustGroup serverTrustGroup = sqlClient.ServerTrustGroups.CreateOrUpdate(resourceGroup.Name, TestEnvironmentUtilities.DefaultLocationId, stgName, parameters);
+				Assert.NotNull(serverTrustGroup);
```
Consider providing a message in Assert.NotNull, as it can be useful. Same goes for other Assert.NotNull and Assert.True appearances.

Commit:
new test
